Gist/Managers/MessageQueueManager.swift

```diff
@@ -2,61 +2,69 @@ import Foundation
 
 class MessageQueueManager {
     private var queueTimer: Timer!
-
+    
     func setup() {
         queueTimer = Timer.scheduledTimer(timeInterval: 10,
                                           target: self,
                                           selector: #selector(checkForMessages),
                                           userInfo: nil,
                                           repeats: true)
-        checkForMessages()
+        
+        // Wait 1 second for app to become active.
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            self.checkForMessages()
+        }
     }
 
     @objc
     private func checkForMessages() {
-        Logger.instance.info(message: "Checking Gist queue service")
-        if let userToken = UserManager().getUserToken() {
-            QueueManager(siteId: Gist.shared.siteId, dataCenter: Gist.shared.dataCenter)
-                .fetchUserQueue(userToken: userToken, completionHandler: { response in
-                switch response {
-                case .success(let responses):
-                    Logger.instance.info(message: "Gist queue service found \(responses.count) new messages")
-                    for queueMessage in responses {
-                        let message = queueMessage.toMessage()
-                        let position = message.gistProperties.position
-
-                        if let routeRule = message.gistProperties.routeRule {
-                            let cleanRouteRule = routeRule.replacingOccurrences(of: "\\", with: "/")
-                            if let regex = try? NSRegularExpression(pattern: cleanRouteRule) {
-                                let range = NSRange(location: 0, length: Gist.shared.getCurrentRoute().utf16.count)
-                                if regex.firstMatch(in: Gist.shared.getCurrentRoute(), options: [], range: range) == nil {
-                                    Logger.instance.debug(message:
-                                        "Current route is \(Gist.shared.getCurrentRoute()), needed \(cleanRouteRule)")
+        if UIApplication.shared.applicationState != .background {
```

Comment:
I would test this code under an iOS app that uses Scenes and a separate test for a separate app using SwiftUI. 

I am not very experienced with Scene apps, but with my SwiftUI experience, if you have built an app using SwiftUI and no longer have an `AppDelegate` file, many functions that you call against `UIApplication.shared` do not work as expected. 

So, I am thinking that *maybe* a SwiftUI app will make this line of code always return false.

Reply:
That's my main worry, we will need to test this fix on a pure SwiftUI app

Reply:
I did a quick check to see if I could find anything on a SwiftUI method to detect application state. My thought is that you could change your if logic to check for a UIKit value and a SwiftUI kit value to see if either of them suggest the app is in the foreground and if so, run the queue. 

Many suggestions I found online were about how to *observe* the app state (not a surprise since it's SwiftUI). I *think* I found something that isn't observing, but just a synchronous check for the application state. 

I used [this article](https://betterprogramming.pub/swiftui-tips-detecting-a-swiftui-apps-active-inactive-and-background-state-a5ff8acf5db1) for inspiration which led me to:

```swift
Environment(EnvironmentValues.scenePhase).wrappedValue
```

Taking [the init function of `Environment`](https://developer.apple.com/documentation/swiftui/environment/init(_:)) to read [the app state](https://developer.apple.com/documentation/swiftui/scenephase) via [`EnvironmentValues`](https://developer.apple.com/documentation/swiftui/environmentvalues).

Reply:
Looking at Apple's [documentation](https://developer.apple.com/documentation/uikit/uiapplication/1623003-applicationstate) about this subject, it looks like what we have should be fine already:

> In a scene-based app, this property takes the value of the most active scene, which it determines from each scene’s [activationState](https://developer.apple.com/documentation/uikit/uiscene/3197908-activationstate) property.

The demo app uses scenes (thinking about it now, that's why I had to add the delay), I'm going to test on a pure "no app delegate" iOS app and revert back, that's the only way to confirm.

Reply:
Hey @levibostian, I created a new, from scratch Swift UI demo app, initialized the SDK in the new "App" file.

```swift
import SwiftUI
import Gist

@main
struct PureSwiftUITestApp: App {
    init() {
        Gist.shared.setup(siteId: "a5ec106751ef4b34a0b9", dataCenter: "eu", logging: true, env: .production)
        Gist.shared.setUserToken("ABC123")
    }
    
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}
```

I confirm that the fix is working as expected, the queue stops when the app is in the background and resumes when the app is active again.

Shall we merge this?